naga/src/back/msl/writer.rs

```diff
@@ -3367,26 +3368,6 @@ impl<W: Write> Writer<W> {
             } = *expr
             {
                 match fun {
-                    crate::MathFunction::Dot => {
-                        // WGSL's `dot` function works on any `vecN` type, but Metal's only
-                        // works on floating-point vectors, so we emit inline code for
-                        // integer vector `dot` calls. But that code uses each argument `N`
-                        // times, once for each component (see `put_dot_product`), so to
-                        // avoid duplicated evaluation, we must bake integer operands.
-
-                        // check what kind of product this is depending
-                        // on the resolve type of the Dot function itself
```

Comment:
I think it's worth preserving this comment in some form for the `Dot4U8Packed` and `Dot4I8Packed` case, where it's important that the arguments get baked both when we use the polyfill (because of the duplicate evaluation issue), and when we don't use the polyfill (because we need them to be emitted before casting to packed chars -- see the comment at the call to `put_casting_to_packed_chars`).